verification/client.go

```diff
@@ -50,7 +101,8 @@ func (c *Client) InitializeContext(cmd *InitCtxCmd) (*InitCtxResp, error) {
 	return &respStruct, nil
 }
 
-func (c *Client) GetProfile() (*GetProfileResp, error) {
+// getProfile is an internal helper for handling GetProfile as part of either the client API or initialization.
+func getProfile(t Transport) (*GetProfileResp, error) {
```

Comment:
Do you think it would be worth while to change this or add another function to be a factory for `Client`. In the tests we could call the factory with the transport and it would return the generic version of the client instead of calling `NewClient256` directly.

I guess in this case we would need to add a function to the `TestDPEInstance` that returns the expected profile to make sure we got what we expected, so maybe it doesn't buy too much for us.

Reply:
I spent a lot of time trying to think of a nice way to do this. Go's generics are quite limited by roughly "what is possible without making the compiler do a lot of work." So, we can have a factory function, but its return type would be `interface{}` (aka any) because there is not a useful interface that covers both flavors of `Client[C, D]`. (A type constraint could be written as `interface { Client[X1, X2] | Client[Y1, Y2] }`, but that is not a valid return type).

So, the approach I've taken here is to require the caller to have an opinion about what instantiation of Client they expect to get. They will get an error if they provide the wrong one. It's not ideal, maybe we can make it nicer in the future.

Reply:
Okay, that seems good to me. Maybe `testCertifyKey` can have a parameter later on for which `Client` to create.